src/styles/_input.scss

```diff
@@ -28,6 +28,40 @@
  * @licence Simplified BSD License
  */
 
+.osjs-gui-field-label {
```

Comment:
I suggest the following changes:

1. Force inner `osjs-gui` elements to not have margins by default
2. Instead of creating surroinding margin, place it in the direction of the label
3. Set label as block and remove width calculation
4. Force `osjs-gui` elements to flex `1 1` when it is placed on left

```css
.osjs-gui-field-label {
  $half-margin: $base-margin / 2;

  display: flex;
  flex-wrap: nowrap;
  justify-content: left;

  & > label {
    display: block;
  }

  & > .osjs-gui {
    margin: 0;
  }

  &.osjs-gui-field-label-on-top {
    flex-direction: column;

    & > label {
      margin-bottom: $half-margin;
    }
  }

  &.osjs-gui-field-label-on-left {
    flex-direction: row;
    align-items: center;

    & > label {
      margin-right: $half-margin;
    }

    & > .osjs-gui {
      flex: 1 1;
    }
  }

  &.osjs-gui-field-label-on-bottom {
    flex-direction: column-reverse;

    & > label {
      margin-top: $half-margin;
    }
  }
}
```

Reply:
Yes, if these changes fit better with the project's current standard I definitely agree with them. However, this removes right and left margins for all label components. Is this what you wanted? I attached a screenshot of what it looks like.
![image](https://user-images.githubusercontent.com/22305849/55245766-2b43c600-521a-11e9-9678-1eb2401c09f8.png)

Reply:
What root element are you using ?

Because if you use a Box (or derivative) it should look OK.

Reply:
> Because if you use a Box (or derivative) it should look OK.

 Indeed it does :)

```javascript
      const view = (state, actions) => h(Box, {
        grow: 1,
        shrink: 1
      }, [
        h(Label, {text: 'Top'}, h(TextField)),
        h(Label, {text: 'Left', placement: 'left'}, h(TextField)),
        h(Label, {text: 'Bottom', placement: 'bottom'}, h(TextField)),
        h(Box, {}, [
          h(Label, {text: 'Nested Top'}, h(TextField))
        ])
      ]);
```

![OS js](https://user-images.githubusercontent.com/161548/55251528-1afb0d80-5251-11e9-9659-4337b66dfbf8.png)

Reply:
Jeez -- I wrote this a while ago, but by accident started a new review :man_facepalming:

Reply:
Yes, that makes sense. I should have looked at the standard in the example applications.